src/clj/kixi/hecuba/data/measurements/calculations.clj

```diff
@@ -242,3 +244,119 @@
   (let [{:keys [start-date end-date]} range]
     (doseq [timestamp (time/seq-dates start-date end-date (t/days 1))]
       (calculate-batch store sensor (t/minus timestamp (t/weeks 4)) (t/plus timestamp (t/days 1)) calculation-type))))
+
+(defn read-tariffs
```

Comment:
Sorry, you are right. I keep getting thown off by the db/with-session stuff that we end up doing outside the call.